scripts/jest/matchers/toWarnDev.js

```diff
@@ -23,16 +24,28 @@ const createMatcherFor = consoleMethod =>
       const warningsWithoutComponentStack = [];
       const warningsWithComponentStack = [];
       const unexpectedWarnings = [];
+      let lastWarningWithMismatchingFormat = null;
 
       // Catch errors thrown by the callback,
       // But only rethrow them if all test expectations have been satisfied.
       // Otherwise an Error in the callback can mask a failed expectation,
       // and result in a test that passes when it shouldn't.
       let caughtError;
 
-      const consoleSpy = message => {
+      const consoleSpy = (format, ...args) => {
+        const message = util.format(format, ...args);
         const normalizedMessage = normalizeCodeLocInfo(message);
 
+        let argIndex = 0;
+        format.replace(/%s/g, () => argIndex++);
+        if (argIndex !== args.length) {
+          lastWarningWithMismatchingFormat = {
+            format,
+            args,
+            expectedArgCount: argIndex,
+          };
+        }
```

Comment:
Clever!

Could use an inline comment. Wasn't clear (to me at least) on first read what it was doing.